### PR TITLE
fixed [FragLand#216](FragLand#216) for javascript, PHP and Python

### PR DESCRIPTION
**Pull request by molanp**
fixed [FragLand#216](FragLand#216) for javascript, PHP and Python

### CODE REVIEW

**Review by ldilley — COMMENTED** [review]
Greetings, @molanp.

In reference to issue #216, only data returned from SLP 1.7 JSON queries is potentially impacted, correct? If this is the case, then the JavaScript changes in this PR are unnecessary considering that the JavaScript implementation only supports SLP 1.4/1.5. Furthermore, the included PHP modifications would not correct the issue for SLP 1.7. The changes would need to be made to the following lines (613 and 614) in `minestat.php`:
```php
      $this->current_players = (int)@$json_data['players']['online'];
      $this->max_players = (int)@$json_data['players']['max'];
```

Tagging @mindsolve to see if he has any input for the proposed Python changes.

File: Python/minestat/__init__.py [hunk 1]
```diff
@@ -797,12 +797,13 @@ def __parse_json_payload(self, payload_raw: Union[bytes, bytearray]) -> ConnStat
...
-    self.current_players = payload_obj["players"]["online"]
+    players = payload_obj.get("players", {"max":-1, "online":-1})
+    self.max_players = players["max"]
+    self.current_players = players["online"]
 
     # There may be a "sample" field in the "players" object that contains a sample list of online players
-    if "sample" in payload_obj["players"]:
-      self.player_list = [player["name"] for player in payload_obj["players"]["sample"]]
+    if "sample" in players:
+      self.player_list = [player["name"] for player in players["sample"]]
```
Comment (mindsolve) [inline Python/minestat/__init__.py:806]: I think this only works if the whole "players" object is missing. If either the "max" or "online" properties are missing but the players element exists, it is still going to raise an exception here. I would prefer if that would also be cought.

**Review by mindsolve — CHANGES_REQUESTED** [review]
Thanks for the ping, ldilley :)

I just have one small comment for the python edition.
Thank you, @molanp for the contribution!

**Comment by molanp** [comment]
Done.

**Review by mindsolve — APPROVED** [review]
In regards to the Python parts: Approved, thanks for the addition!

**Review by ldilley — APPROVED** [review]
The JavaScript and PHP changes now look good as well. Thank you for the fixes, @molanp!

**Pull request merged**